ledger/eval/eval.go

```diff
@@ -1637,18 +1742,25 @@ func (eval *BlockEvaluator) generateKnockOfflineAccountsList() {
 					updates.ExpiredParticipationAccounts,
 					accountAddr,
 				)
-				continue // if marking expired, do not also suspend
+				continue // if marking expired, do not consider suspension
 			}
 		}
 
+		// Absent check: has it been too long since the last heartbeat/proposal, or
+		// has this online account failed a challenge?
 		if len(updates.AbsentParticipationAccounts) >= maxSuspensions {
 			continue // no more room (don't break the loop, since we may have more expiries)
 		}
 
-		if acctData.Status == basics.Online {
+		if acctData.Status == basics.Online && acctData.IncentiveEligible {
```

Comment:
We landed on have to be incentive eligible to be knocked offline for this release.